posts/immune-cell-typing-smooth/index.qmd

```diff
@@ -0,0 +1,1060 @@
+
+```{r, eval=FALSE}
+
+tcell_ids <- sem@meta.data[sem@meta.data$clusters_unsup==6,"cell_ID"]
+metagenes_t <- 
+  fit_metagene_scores(
+    markerslist = tcell_markerlist
+    ,counts_matrix = Matrix::t(sem[["RNA"]]@counts[,tcell_ids])
+    ,adjacency_matrix = as(sem@graphs$umapgrph
+                           ,"CsparseMatrix")[tcell_ids,tcell_ids]
+  )
+```
+
+The `metagenes_t` object we've created has an element for each celltype, including the regression coefficients (`bhat`), and predicted values we'll use for clustering (`yhat`) for each cell and each cell type.
+
+```{r}
+str(metagenes_t,max.level = 2)
+```
+
+
+We can get a better understanding of the model output by looking at a few diagnostics plots.  In the below, we plot the regression coefficients for each model, ordering them on the x-axis by their ranking.
+For each cell type, the positive $\hat{\beta}$ correspond to genes we specified as predictors, and negative $\hat{\beta}$ were predictors for either of the other two cell types.  
+A lot of the covariates which have the largest magnitudes have "*lag.*" in their name indicating they correspond to a smoothed (or neighbor-averaged) covariate for that gene.
+
+For `cd4t`, we see that smoothed CD4 (indicated by '*lag.y*' in the plot) is the strongest predictor of observed 'CD4'.  For `cd8t` and `treg` categories, '*lag.y*' (indicating smoothed CD8B and FOXP3) are also among the strongest predictors, but not first in rank.
+
+
+
+```{r, eval=FALSE}
+
+coefplots <- metagene_coefficients_plot(metagenes_t)
+coefplots <- lapply(coefplots, function(x){x +   theme(text=element_text(face='bold',size=18))
+  }) ## just make the text bigger and bold
+cowplot::plot_grid(plotlist = coefplots, nrow = 1)
+
+```
+
+<div class="wider-image">
+  <img src="./figures/coefplots_tcells.png" alt="Wide Image">
+</div>
+
+```{r}
+#| eval: false
+#| echo: false
+
+knitr::include_graphics("./figures/coefplots_tcells.png")
+```
+
+A pairsplot can show us how well the metagene scores we've created are distinct from each other.  Ideally, if our cell type classes are mutually exclusive, we'd want to not see too many cells which are 'positive' for multiple cell types at the same time.
+
+Here, we see that `cd4t` and `treg` scores are positively correlated.  This isn't too surprising, because T-reg cells are actually a subtype of CD4+ T cells.
+`cd8t` is somewhat negatively correlated with both `treg` and `cd4t`, which is good to see.
+
+```{r, eval=FALSE}
+metagene_pairsplot(metagenes_t, var= "yhat")
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/yhatpairs_tcells.png")
+```
+
+### Part 3: Clustering via overfitted gaussian mixture-models 
+
+Finally, we will cluster these scores to infer the T cell types.
+The `cluster_metagenes` function we run below estimates a mixture of overfitted multivariate gaussian mixture models [@aragam2020identifiability] to estimate the distribution of metagene scores.
```

Comment:
give some intuition here. something like, "now, for each cell type, we estimate the multivariate distribution of metagene scores it falls within"